daemon/src/entity/commands/pty/pty_start.ts

```diff
@@ -194,38 +193,70 @@ export default class PtyStartCommand extends AbsStartCommand {
       pipeName = `\\\\.\\pipe\\mcsmanager-${pipeId}`;
     }
 
+    // Get user info for the target user (Linux/macOS only)
+    let uid: number | undefined;
+    let gid: number | undefined;
+    if (process.platform !== 'win32' && instance.config.runAs) {
+      try {
+        const { execSync } = require('child_process');
+        uid = parseInt(execSync(`id -u ${instance.config.runAs}`).toString().trim());
+        gid = parseInt(execSync(`id -g ${instance.config.runAs}`).toString().trim());
+        
+        // Ensure working directory has correct permissions
+        fs.chownSync(instance.absoluteCwdPath(), uid, gid);
+        commandList = commandStringToArray("sudo" + " -u " + instance.config.runAs);
```

Comment:
I believe the sudo command is unreliable, as it may not exist in the Linux system or Docker container where MCSManager is running. You should use the same method as in `general_start.ts` to set the program user: 

"pty / pty.exe" is actually a subprocess, and I think it should be possible to assign different users in the same way.

Then you’ll find that you’ve written a lot of duplicate code in the two TypeScript files. You should extract this duplicate code into a new file to facilitate code reuse and improve code robustness.